### PR TITLE
API Deprecate campaign admin integrations

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripeltd/product-issues/issues/884

### CODE REVIEW

File: code/Extensions/CampaignAdminExtension.php [hunk 1]
```diff
@@ -7,15 +7,23 @@
...
  * This extension will only be applied if the `campaign-admin` module is installed.
  *
  * @extends Extension<FileFormFactory>
+ * @deprecated 5.4.0 Will be removed without equivalent functionality to replace it
  */
 class CampaignAdminExtension extends Extension
 {
+    public function __construct()
+    {
+        Deprecation::notice('5.4.0', 'Will be removed without equivalent functionality to replace it', Deprecation::SCOPE_CLASS);
```
Comment (GuySartorelli) [inline code/Extensions/CampaignAdminExtension.php:23]: Use the `noticeWithoutReplacement` so it gets wrapped. Anything without a replacement should be wrapped.

Comment (ChloeSartorelli) [inline code/Extensions/CampaignAdminExtension.php]: Use the `noticeWithNoReplacment` so it gets wrapped. Anything without a replacement should be wrapped.
```suggestion
       Deprecation::noticeWithNoReplacment('5.4.0', scope: Deprecation::SCOPE_CLASS);
```

**Review by ChloeSartorelli — CHANGES_REQUESTED** [review]

File: code/Extensions/CampaignAdminExtension.php (hunk 1 above)
Comment (ChloeSartorelli) [inline code/Extensions/CampaignAdminExtension.php]:
```suggestion
 * @deprecated 2.4.0 Will be removed without equivalent functionality to replace it
 */
class CampaignAdminExtension extends Extension
{
    public function __construct()
    {
        Deprecation::notice('2.4.0', 'Will be removed without equivalent functionality to replace it', Deprecation::SCOPE_CLASS);
```

**Review by ChloeSartorelli — CHANGES_REQUESTED** [review]

File: code/Extensions/CampaignAdminExtension.php (hunk 1 above)
Comment (ChloeSartorelli) [reply code/Extensions/CampaignAdminExtension.php]: `2.4.0` not `5.4.0`

Comment (emteknetnz) [reply code/Extensions/CampaignAdminExtension.php]: Updated

Comment (emteknetnz) [reply code/Extensions/CampaignAdminExtension.php]: Updated

Comment (ChloeSartorelli) [reply code/Extensions/CampaignAdminExtension.php]: You missed the PHPDoc

Comment (emteknetnz) [reply code/Extensions/CampaignAdminExtension.php]: Opps, done

**Review by ChloeSartorelli — APPROVED** [review]
LGTM

**Pull request merged**